src/dmd/dstruct.d

```diff
@@ -445,7 +445,7 @@ extern (C++) class StructDeclaration : AggregateDeclaration
         if (!elements)
             return true;
 
-        size_t nfields = fields.dim - isNested();
```

Comment:
This expression is used https://github.com/dlang/dmd/blob/7c827fdf6d503d3e75197223896f917315689396/src/dmd/initsem.d#L124 which is untouched by this PR, make it a function and use it everywhere?

Reply:
what do you mean?

Reply:
edit

Reply:
> then the code would become `sd.fields.dim - sd.hiddenFields`

I was thinking more like `const nfields = sd.nonHiddenFields();`.

> but I don't see it complicated enough already to go to a function.

I agree that it is trivial, but complexity (or lack thereof) has nothing to do with it, its about readability and copyability. Having one way to doing it prevents accidentally omitting one of them. Choosing a good name for it also makes it immediately obvious what is going on.

Reply:
done https://github.com/dlang/dmd/pull/9282/commits/80eb2f4b60e4d78cf8c3c0b04b51dd2af132ab9e.

Reply:
Thanks!